br/pkg/lightning/backend/kv/sql2kv.go

```diff
@@ -440,12 +440,20 @@ func isPKCol(colInfo *model.ColumnInfo) bool {
 	return mysql.HasPriKeyFlag(colInfo.GetFlag())
 }
 
-func (kvcodec *tableKVEncoder) getActualDatum(rowID int64, inputDatum *types.Datum, col *table.Column) (types.Datum, error) {
+func (kvcodec *tableKVEncoder) getActualDatum(rowID int64, colIndex int, inputDatum *types.Datum) (types.Datum, error) {
 	var (
 		value types.Datum
 		err   error
 	)
+
 	tblMeta := kvcodec.tbl.Meta()
+	cols := kvcodec.tbl.Cols()
+
+	if colIndex < 0 || colIndex >= len(cols) {
```

Comment:
Why change `cols` to `colIndex`? It introduces redundant checks.

Reply:
Because the kvcodec contains the table and the columns itself,  passing an index can prevent the caller using an arbitrary column not belonging to the table as an argument.

Reply:
since it's a internal function, we can remove this check